packages/cli/src/webhooks/WebhookRequestHandler.ts

```diff
@@ -0,0 +1,122 @@
+import type express from 'express';
+import type { IHttpRequestMethods } from 'n8n-workflow';
+import type { IWebhookManager, WebhookCORSRequest, WebhookRequest } from '@/webhooks/webhook.types';
+import * as ResponseHelper from '@/ResponseHelper';
+
+export const WEBHOOK_METHODS: IHttpRequestMethods[] = [
+	'DELETE',
+	'GET',
+	'HEAD',
+	'PATCH',
+	'POST',
+	'PUT',
+];
+
+class WebhookRequestHandler {
+	constructor(private readonly webhookManager: IWebhookManager) {}
+
+	/**
+	 * Handles an incoming webhook requests. Handles CORS and delegates the
+	 * request to the webhook manager to execute the webhook.
+	 */
+	async handleRequest(req: WebhookRequest | WebhookCORSRequest, res: express.Response) {
```

Comment:
`WebhookCORSRequest` → CORS also applies to requests that are not preflighted, but `WebhookCORSRequest` refers only to `OPTIONS` ones. Maybe this would be clearer as `WebhookPreflightedRequest` or simply `OptionsRequest`.